packages/renderer/src/lib/preferences/PreferencesRenderingItemFormat.svelte

```diff
@@ -58,28 +75,32 @@ onDestroy(() => {
   }
 });
 
-$: if (resetToDefault) {
-  recordValue = record.type === 'number' ? getNormalizedDefaultNumberValue(record) : record.default;
-  if (ensureType(recordValue)) {
-    update(record).catch((err: unknown) => console.log(`Error updating ${record.id}`, err));
-  }
-
-  resetToDefault = false;
-}
+$effect(() => {
+  if (resetToDefault) {
+    recordValue = record.type === 'number' ? getNormalizedDefaultNumberValue(record) : record.default;
+    if (ensureType(recordValue)) {
+      update(record).catch((err: unknown) => console.log(`Error updating ${record.id}`, err));
+    }
 
-$: if (!isEqual(currentRecord, record)) {
-  initialValue
-    .then(value => {
-      recordValue = value;
-      if (record.type === 'boolean') {
-        recordValue = !!value;
-      }
-    })
-    .catch((err: unknown) => console.error('Error getting initial value', err));
+    resetToDefault = false;
+  }
+});
 
-  invalidText = undefined;
-  currentRecord = record;
-}
+$effect(() => {
+  if (!isEqual(currentRecord, record)) {
+    initialValue
+      .then(value => {
+        recordValue = value;
+        if (record.type === 'boolean') {
+          recordValue = !!value;
+        }
+      })
+      .catch((err: unknown) => console.error('Error getting initial value', err));
+
+    invalidText = undefined;
+    currentRecord = record;
```

Comment:
I know this is refactoring, and not related to the code logic, but I don't understand the code logic here :/